src/tests/mod.rs

```diff
@@ -90,3 +91,9 @@ pub fn sizecbuf<'a>(size: usize) -> Box<[u8]> {
 pub fn cbuf2str(buf: &[u8]) -> &str {
     std::str::from_utf8(buf).unwrap()
 }
+
+pub fn generate_random_port() -> u16 {
```

Comment:
add a comment here quickly explaining why were randomizing ports and in what situations we do so

Reply:
This is useful, but doesn't guarantee these are unused.  There are ways to do this: https://superuser.com/questions/885414/linux-command-get-unused-port

I think random is better than fixed, but checking for a port to be unused may be better than either...

Reply:
To expand on this, you have a ~2.5% chance of a collision now (just from your own tests), and if we get to 50 tests, it'll be ~7.5%.  This is enough that we will frequently see errors when testing and so will be conditioned that false alarms are expected.  I'd prefer to sidestep this, if possible, especially if we're doing automated testing on every commit...

Reply:
I initially underestimated the likelihood of collisions. However, your calculations demonstrate the necessity of checking for unused ports. I have updated the function accordingly.